antipackage.py

```diff
@@ -1,130 +1,401 @@
+            self._install_init()
+
+    def _install_init(self):
+        if self.path != _BASE_DIR:
+            self._touch()
 
-    def _touch(self, path):
+    def _touch(self):
+        i = '__init__.py'
+        path = os.path.join(self.path, i)
         with open(path, 'a'):
             os.utime(path, None)
 
-    def _install_module(self, fullname):
-        top, username, repo, modname = self._parse_fullname(fullname)
-        url = 'https://raw.githubusercontent.com/%s/%s/master/%s' % (username, repo, modname+'.py')
-        print('Downloading: ', url)
-        try:
-            tmp_file, resp = urlretrieve(url)
-            with open(tmp_file, 'r') as f:
-                new_content = f.read()
-            if new_content=='Not Found':
-                raise InstallError('remote file does not exist')
-        except IOError:
-            raise InstallError('error downloading file')
-        
-        new = tmp_file
-        old = self._install_path(fullname)
-        updated = self._update_if_changed(old, new)
-        if updated=='updated':
-            print('Updating module: ', fullname)
-        elif updated=='installed':
-            print('Installing module: ', fullname)
-        elif updated=='noaction':
-            print('Using existing version: ', fullname)
-
-    def _install_path(self, fullname):
-        top, username, repo, modname = self._parse_fullname(fullname)
-        return os.path.join(self.base_dir, top, username, repo, modname+'.py')
-
-    def _make_package(self, fullname):
-        top, username, repo, modname = self._parse_fullname(fullname)
-        if repo is not None:
-            repo_path = os.path.join(self.base_dir, top, username, repo)
-            self._setup_package(repo_path)
-        if username is not None:
-            user_path = os.path.join(self.base_dir, top, username)
-            self._setup_package(user_path)
-        if top is not None:
-            top_path = os.path.join(self.base_dir, top)
-            self._setup_package(top_path)
-        if modname is not None:
+    def waiting(self):
+        """Ready for import?"""
+        if not self.repo or len(self.comps)>3:
+            return True
+        return False
+
+    def exists(self):
+        """Data and self.path exist?"""
+        self.data = data('/'.join(self.index[1:]))
+        return os.path.exists(self.path) and self.data
+
+    def update(self):
+        """Use existing files?"""
+        pathlist = self.index[1:]
+        reponame = '.'.join(pathlist)
+        for key in self.data:
+            if key in ('branch','tag'):
+                repo = GitHubRepo(pathlist, key, self.data[key])
+                if repo.sha != self.data['commit']['sha']:
+                    print('Updating repo: %s' % reponame)
+                    self.data['commit']['sha'] = repo.sha
+                    return True
+        print('Using existing version of: %s' % reponame)
+        return False
+
+    def install(self):
```

Comment:
Let's move the install method to the `GitHubRepo` and give it the signature `download(self, path)` where `path` is the location the directory will end up. The `_install_init` should not be done in that new method though.